sg: Add `sg rfc [list|search|open]` commands

mrnugget opened the pull request:
Demo video:

https://user-images.githubusercontent.com/1185253/125973447-ee13e2be-b644-44ae-a910-7225cf13403b.mp4

jeanduplessis commented on dev/sg/rfc.go:
```diff
@@ -0,0 +1,210 @@
+	return nil
+}
+
+func openURL(url string) error {
```
We should consider extracting this into a utils package at some point

jeanduplessis commented on dev/sg/rfc.go:
```diff
@@ -0,0 +1,227 @@
+		Q(q).
+		Fields("nextPageToken, files(id, name, parents)")
+
+	if orderBy != "" {
```
FYI: When querying over the fullText field, as in the `searchRFCs` function, you can't use orderBy which is why I added this check.

mrnugget commented:
@jeanduplessis I fixed the issue of `sg rfc [...]` not working outside `sourcegraph/sourcegraph`. I removed the implicit call to the `root.RepositoryRoot` function in `internal/db/db.go` and added explicit calls to `parseConf` to all commands that need it. That's a bit of boilerplate that we will probably need to clean up in the future. But it's fine for now.

mrnugget commented:
@tjdevries tagging you just for an FYI that I touched your `internal/db/db.go` code a bit.

tjdevries commented on dev/sg/rfc.go:
```diff
@@ -0,0 +1,240 @@
+
+const (
+	tokenFileName = ".sg.token.json"
+	credentials   = `{"installed":{"client_id":"1043390970557-1okrt0mo0qt2ogn2mkp217cfrirr1rfd.apps.googleusercontent.com","project_id":"sg-cli","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"gkQ2alKQZr2088IFGr55ET_I","redirect_uris":["urn:ietf:wg:oaut ...
```
what is this?

mrnugget replied:
Static oauth2 client data. See [this query](https://sourcegraph.com/search?q=context:global+%22token_uri%22:%22https://oauth2.googleapis.com/token%22&patternType=literal) :smile:

tjdevries commented:
db changes seem fine, thanks :)
